test/FunctionalTests/Microsoft.OData.Core.Tests/ODataWriterValidationTests.cs

```diff
@@ -0,0 +1,3356 @@
+                    new ODataProperty { Name = "Amount", Value = 130M }
+                }
+            };
+
+            this.ordersNestedResourceInfo = new ODataNestedResourceInfo
+            {
+                Name = "Orders",
+                IsCollection = true,
+                Url = new Uri("http://tempuri.org/Customers(1)/Orders")
+            };
+
+            this.locationResource = new ODataResource
+            {
+                TypeName = "NS.Address",
+                Properties = new List<ODataProperty>
+                {
+                    new ODataProperty { Name = "Street", Value = "Waiyaki Way" }
+                }
+            };
+
+            this.officeLocationsNestedResourceInfo = new ODataNestedResourceInfo
+            {
+                Name = "OfficeLocations",
+                IsCollection = true,
+                IsComplex = true
+            };
+
+            this.customerNestedResourceInfo = new ODataNestedResourceInfo
+            {
+                Name = "Customer",
+                IsCollection = false,
+                Url = new Uri("http://tempuri.org/Orders(1)/Customer")
+            };
+
+            this.shippingAddressNestedResourceInfo = new ODataNestedResourceInfo
+            {
+                Name = "ShippingAddress",
+                IsCollection = false,
+                IsComplex = true
+            };
+
+            this.shippingAddressResource = new ODataResource
+            {
+                TypeName = "NS.Address",
+                Properties = new List<ODataProperty>
+                {
+                    new ODataProperty { Name = "Street", Value = "Wood Avenue" }
+                }
+            };
+
+            var productPhotoMediaResource = new ODataStreamReferenceValue
+            {
+                EditLink = new Uri($"http://tempuri.org/Products(1)/Photo"),
+                ReadLink = new Uri($"http://tempuri.org/Products(1)/Photo"),
+                ContentType = "image/png",
+                ETag = "media-etag"
+            };
+
+            this.productResource = new ODataResource
+            {
+                TypeName = "NS.Product",
+                Properties = new List<ODataProperty>
+                {
+                    new ODataProperty { Name = "Id", Value = 1 },
+                    new ODataProperty { Name = "Name", Value = "Lenovo ThinkPad P53s" }
+                },
+                MediaResource = new ODataStreamReferenceValue()
+            };
+
+            this.productResource.MediaResource.SetMetadataBuilder(
+                new TempEntityMetadataBuilder(
+                    resourceId: this.productResource.Id,
+                    mediaResource: productPhotoMediaResource,
+                    eTag: null),
+                "Photo");
+
+            this.messageWriterSettings = new ODataMessageWriterSettings
+            {
+                EnableMessageStreamDisposal = false,
+                Version = ODataVersion.V4,

```

Comment:
Will all the tests work with `ODataVersion.V401`

Reply:
@WanjohiSammy Parameterized all tests such that they're ran for both V4 and V401.